backend-api/src/functions/asyncFinishBiometricSession/asyncFinishBiometricSessionHandler.ts

```diff
@@ -23,20 +35,181 @@ export async function lambdaHandlerConstructor(
   logger.resetKeys();
   logger.addContext(context);
   logger.info(LogMessage.FINISH_BIOMETRIC_SESSION_STARTED);
+  const configResult = getFinishBiometricSessionConfig(dependencies.env);
+  if (configResult.isError) {
+    return serverErrorResponse;
+  }
+
+  const config = configResult.value;
 
-  const validateRequestBodyResult = validateRequestBody(event.body);
-  if (validateRequestBodyResult.isError) {
-    const errorMessage = validateRequestBodyResult.value.errorMessage;
+  const validateResult = validateRequestBody(event.body);
+
+  if (validateResult.isError) {
     logger.error(LogMessage.FINISH_BIOMETRIC_SESSION_REQUEST_BODY_INVALID, {
-      errorMessage,
+      errorMessage: validateResult.value.errorMessage,
     });
-    return badRequestResponse("invalid_request", errorMessage);
+    return badRequestResponse(
+      "invalid_request",
+      validateResult.value.errorMessage,
+    );
+  }
+
+  const { sessionId, biometricSessionId } = validateResult.value;
+  const eventService = dependencies.getEventService(config.TXMA_SQS);
+  const sessionRegistry = dependencies.getSessionRegistry(
+    config.SESSION_TABLE_NAME,
+  );
+
+  const updateResult = await sessionRegistry.updateSession(
+    sessionId,
+    new BiometricSessionFinished(biometricSessionId),
+  );
+
+  if (updateResult.isError) {
+    return handleUpdateSessionError(
+      updateResult,
+      eventService,
+      sessionId,
+      biometricSessionId,
+      config.ISSUER,
+    );
   }
 
   logger.info(LogMessage.FINISH_BIOMETRIC_SESSION_COMPLETED);
   return notImplementedResponse;
 }
 
+async function handleConditionalCheckFailure(
+  eventService: IEventService,
+  sessionAttributes: SessionAttributes,
+  biometricSessionId: string,
+  issuer: string,
+): Promise<APIGatewayProxyResult> {
+  const sessionAge = Date.now() - sessionAttributes.createdAt;
+  const isSessionExpired = sessionAge > 60 * 60 * 1000;
+
+  function getFraudSignal(
+    expired: boolean,
+  ): Record<string, string> | undefined {
+    if (expired) {
+      return { suspected_fraud_signal: "AUTH_SESSION_TOO_OLD" };
+    }
+  }
```

Comment:
```suggestion
  function getFraudSignal(
    expired: boolean,
  ): Record<string, string> | undefined {
    if (!expired) {
      return undefined
    }
    
    return { suspected_fraud_signal: "AUTH_SESSION_TOO_OLD" };
  }
```

Purely preference, I prefer to see a declarative return of undefined but please resolve if you don't agree